src/fetch/orderbook.py

```diff
@@ -0,0 +1,217 @@
+        res = []
+        while start <= end:
+            size = min(end - start + 1, bucket_size)
+            log.info(f"About to process block range ({start}, {start + size - 1})")
+            res.append(
+                cls.run_batch_data_query(
+                    BlockRange(block_from=start, block_to=start + size - 1),
+                    config=config,
+                )
+            )
+            start = start + size
+        return pd.concat(res)
+
+    @classmethod
+    def run_order_data_query(cls, block_range: BlockRange) -> DataFrame:
+        """
+        Fetches and validates Order Data DataFrame as concatenation from Prod and Staging DB
+        """
+        cow_reward_query_prod = (
+            open_query("orderbook/order_data.sql")
+            .replace("{{start_block}}", str(block_range.block_from))
+            .replace("{{end_block}}", str(block_range.block_to))
+            .replace("{{env}}", "prod")
+        )
+        cow_reward_query_barn = (
+            open_query("orderbook/order_data.sql")
+            .replace("{{start_block}}", str(block_range.block_from))
+            .replace("{{end_block}}", str(block_range.block_to))
+            .replace("{{env}}", "barn")
+        )
+        data_types = {"block_number": "int64", "amount": "float64"}
+        barn, prod = cls._query_both_dbs(
+            cow_reward_query_prod, cow_reward_query_barn, data_types
+        )
+
+        # Warn if solver appear in both environments.
+        if not set(prod.solver).isdisjoint(set(barn.solver)):
+            log.warning(
+                f"solver overlap in {block_range}: solvers "
+                f"{set(prod.solver).intersection(set(barn.solver))} part of both prod and barn"
+            )
+
+        if not prod.empty and not barn.empty:
+            return pd.concat([prod, barn])
+        if not prod.empty:
+            return prod.copy()
+        if not barn.empty:
+            return barn.copy()
+        return pd.DataFrame()
+
+    @classmethod
+    def get_order_data(
+        cls, block_range: BlockRange, config: AccountingConfig
+    ) -> DataFrame:
+        """
+        Decomposes the block range into buckets of 10k blocks each,
+        so as to ensure the batch data query runs fast enough.
+        At the end, it concatenates everything into one data frame
+        """
+        load_dotenv()
+        start = block_range.block_from
+        end = block_range.block_to
+        bucket_size = config.data_processing_config.bucket_size
+        res = []
+        while start < end:
+            size = min(end - start + 1, bucket_size)
+            log.info(f"About to process block range ({start}, {start + size - 1})")
+            res.append(
+                cls.run_order_data_query(
+                    BlockRange(block_from=start, block_to=start + size - 1)
+                )
+            )
+            start = start + size
+        return pd.concat(res)
```

Comment:
I know that this is not horrible code duplication. Not many lines can be saved by moving the chunking into its own function and one would have to awkwardly pass a function as argument to it.
But then again the logic of chunking always has potential for off by one errors. So I would still think it is worth it to refactor this at some point into one function for chunking which then can also be properly tested.